Lib/fontTools/varLib/varStore.py

```diff
@@ -217,10 +217,16 @@ def VarStore_subset_varidxes(self, varIdxes, optimize=True):
 
 		items = data.Item
 		newItems = []
-		for minor in sorted(usedMinors):
-			newMinor = len(newItems)
-			newItems.append(items[minor])
-			varDataMap[(major<<16)+minor] = (newMajor<<16)+newMinor
+		if major == 0 and retainFirstMap:
+			zeros = [0] * len(items[0])
+			for minor in range(len(items)):
+				newItems.append(items[minor] if minor in usedMinors else zeros)
```

Comment:
it would be better to append a copy of that `zeros` list instead of a reference to the original (mutable) object. The expectation is that each VarData item is a distinct list object

Reply:
inlined zero array creation